lib/actions/call-taker.js

```diff
@@ -73,9 +74,10 @@ export function endCall () {
       {method: 'POST', body: callData}
     )
       .then(res => res.json())
-      .then(id => {
+      .then(async id => {
         // Inject call ID into active call and save queries.
-        dispatch(saveQueriesForCall({...activeCall, id}))
+        await dispatch(saveQueriesForCall({...activeCall, id}))
+        // Wait until query was saved before re-fetching queries for this call.
         dispatch(fetchCalls())
```

Comment:
I find that this doesn't work for me. Also, if we're introducing async/await, I would prefer that we don't nest that inside calls to `then`.
```javascript
    const id = await fetch(`${otp.config.datastoreUrl}/calltaker/call`,
      {method: 'POST', body: callData}
    )
      .then(res => res.json())
      .catch(err => {
        console.error(err)
        alert(`Could not save call: ${JSON.stringify(err)}`)
      })
    // Inject call ID into active call and save queries.
    await dispatch(saveQueriesForCall({...activeCall, id}))
    // Wait until query was saved before re-fetching queries for this call.
    await dispatch(fetchCalls())
```

Reply:
Refactored in https://github.com/opentripplanner/otp-react-redux/pull/395/commits/789205533d4a85ad0ec30dfe71c19b84f60cd4ae. @landonreed if after this refactor this is still not working for you, could you share the steps or could we have a walkthrough? Thanks.

Reply:
Hopefully this is fixed with https://github.com/opentripplanner/otp-react-redux/pull/395/commits/076454ba7c7d0ba9e43e3338e941e2bd6c396307 and the dev merge.